frontend-react/src/components/FileHandlers/FileHandler.tsx

```diff
@@ -91,25 +106,54 @@ export function getClientHeader(
 
 function FileHandler() {
     const { state, dispatch } = useFileHandler();
-    const [fileContent, setFileContent] = useState("");
-
     const {
         fileInputResetValue,
         contentType,
         fileType,
         fileName,
         errors,
+        errorType,
         destinations,
         reportId,
         successTimestamp,
-        cancellable,
-        errorType,
         warnings,
         localError,
-        overallStatus,
         reportItems,
         selectedSchemaOption,
     } = state;
+    const [fileContent, setFileContent] = useState("");
+    const [fileHandlerStepIndex, setFileHandlerStepIndex] = useState(0);
+    // The File Validate tool now has 4 discrete steps,
+    // Schema Select, File Select, [optional]Show Errors, Success Page
+    // The stages can be seen here: https://figma.fun/fGCeo4
+    const fileHandlerStep = fileHandlerStepsArray[fileHandlerStepIndex];
+    const resetState = () => {
+        setFileContent("");
+        dispatch({ type: FileHandlerActionType.RESET });
+    };
+
+    const handlePrevFileHandlerStep = () => {
+        // We have to clear different parts of state depending
+        // on what step they're coming from. From 3 -> 2, we
+        // need to reset everything EXCEPT the selected schema
+        // since they're just re-selecting a file, otherwise,
+        // we can reset all of state.
+        if (fileHandlerStep === FileHandlerSteps.STEP_TWO) {
+            resetState();
+            setFileHandlerStepIndex(fileHandlerStepIndex - 1);
+        } else if (fileHandlerStep === FileHandlerSteps.STEP_THREE) {
+            resetState();
+            dispatch({
+                type: FileHandlerActionType.SCHEMA_SELECTED,
+                payload: selectedSchemaOption,
+            });
+            setFileHandlerStepIndex(fileHandlerStepIndex - 1);
+        }
+    };
+
+    const handleNextFileHandlerStep = (nextStep = fileHandlerStepIndex + 1) => {
```

Comment:
I don't think we should default to proceeding to the next step because each step is based on the assumptions of each prior step.  This is related to [my comment above](https://github.com/CDCgov/prime-reportstream/pull/8714/files#r1137187764) about computing the index rather than setting it explicitly, though.